### ISSUE TITLE
Research: How to print the data in html from the database using php

### ISSUE THREAD

**Issue by pash0107** [IT3A-2016/Bitsparks#63]
Guys, I need the exact code using php and javascript. Just search "How to print data in html from database using php"

**Comment by pash0107** [comment]
Note: Wag link ang ilagay nio, yung mismong naresearch nio. Slaamat

**Comment by pash0107** [comment]
Noli, Hanap ka ng madaming codes for this one. Para madami tayong options

**Comment by NoliJay** [comment]
<?php
$link = mysqli_connect("localhost", "my_user", "my_password", "world");

/* check connection */

if (mysqli_connect_errno()) {
    printf("Connect failed: %s\n", mysqli_connect_error());
    exit();
}

/* Create table doesn't return a resultset */

if (mysqli_query($link, "CREATE TEMPORARY TABLE myCity LIKE City") === TRUE) {
    printf("Table myCity successfully created.\n");
}

/* Select queries return a resultset */

if ($result = mysqli_query($link, "SELECT Name FROM City LIMIT 10")) {
    printf("Select returned %d rows.\n", mysqli_num_rows($result));

    /* free result set */
    mysqli_free_result($result);
}

/* If we have to retrieve large amount of data we use MYSQLI_USE_RESULT */

if ($result = mysqli_query($link, "SELECT * FROM City", MYSQLI_USE_RESULT)) {

    /* Note, that we can't execute any functions which interact with the
       server until result set was closed. All calls will return an
       'out of sync' error */

    if (!mysqli_query($link, "SET @a:='this will not work'")) {
        printf("Error: %s\n", mysqli_error($link));
    }
    mysqli_free_result($result);
}

mysqli_close($link);
?>

**Comment by NoliJay** [comment]
<?php
$servername = "localhost";
$username = "username";
$password = "password";
$dbname = "myDB";

// Create connection
$conn = mysqli_connect($servername, $username, $password, $dbname);
// Check connection
if (!$conn) {
    die("Connection failed: " . mysqli_connect_error());
}

$sql = "SELECT id, firstname, lastname FROM MyGuests";
$result = mysqli_query($conn, $sql);

if (mysqli_num_rows($result) > 0) {
    // output data of each row
    while($row = mysqli_fetch_assoc($result)) {
        echo "id: " . $row["id"]. " - Name: " . $row["firstname"]. " " . $row["lastname"]. "<br>";
    }
} else {
    echo "0 results";
}

mysqli_close($conn);
?>

**Comment by NoliJay** [comment]
<html>
<head>
<script>
function showUser(str) {
    if (str == "") {
        document.getElementById("txtHint").innerHTML = "";
        return;
    } else { 
        if (window.XMLHttpRequest) {
            // code for IE7+, Firefox, Chrome, Opera, Safari
            xmlhttp = new XMLHttpRequest();
        } else {
            // code for IE6, IE5
            xmlhttp = new ActiveXObject("Microsoft.XMLHTTP");
        }
        xmlhttp.onreadystatechange = function() {
            if (this.readyState == 4 && this.status == 200) {
                document.getElementById("txtHint").innerHTML = this.responseText;
            }
        };
        xmlhttp.open("GET","getuser.php?q="+str,true);
        xmlhttp.send();
    }
}
</script>
</head>
<body>

<form>
<select name="users" onchange="showUser(this.value)">
  <option value="">Select a person:</option>
  <option value="1">Peter Griffin</option>
  <option value="2">Lois Griffin</option>
  <option value="3">Joseph Swanson</option>
  <option value="4">Glenn Quagmire</option>
  </select>
</form>
<br>
<div id="txtHint"><b>Person info will be listed here...</b></div>

</body>
</html>

**Comment by NoliJay** [comment]
<?php
	session_start();

	// Connection Variables
	$hostname="localhost";
	$mysql_login="user";
	$mysql_password="pw";
	$database='sitepoint';
	

	// Connect to the database server
	$link = mysql_connect($hostname, $mysql_login, $mysql_password);
	if (!$link)
	{
		die('Could not connect: ' . mysql_error());
	}
   $db_selected = mysql_select_db($database, $link);
  if (!$db_selected) {
      die ('Can\\'t use ' . $database . ' : ' . mysql_error());
  }

  $myQuery = "SELECT * FROM Meeting_Places ";
			
	if ($_SERVER['REQUEST_METHOD'] == 'POST')
  {
    $task = isset($_POST['task']) ? $_POST['task'] : false;
     if ($task == "Register") {
      $formValues = $_POST;
      $formErrors = array();
      if (!VerifyForm($formValues, $formErrors))
      DisplayForm($formValues, $formErrors);
      else     {
        ProcessForm($formValues);     }
      }
      else if ($task == 'ReturnToMainPage')
        header('Location: index.php');
  }
  else
  {
    DisplayForm(null, null);
}


function DisplayForm($values, $errors)
{ ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"></meta>
<title>SACCC Meeting Information Change</title>
<link rel="stylesheet" type="text/css" href="view.css" media="all" />
<script type="text/javascript" src="view.js"></script>
<script type="text/javascript" src="calendar.js"></script>
</head>
<body id="main_body" >

	<img id="top" src="top.png" alt=""></img>
	<div id="form_container">
	
		<h1><a>SACCC Meeting Information Change</a></h1>
		<form id="form_754506" class="appnitro"  method="post" action="">
					<div class="form_description">
			<h2>SACCC Meeting Information Change</h2>
			<p>Use this form to change the SACCC Monthly Meeting information.</p>
		</div>						
			<ul >		
				<li id="li_2" >
					<label class="description" >Meeting Date: </label>
					<span>
						<input id="element_2_1" name="element_2_1" class="element text" size="2" maxlength="2" value="" type="text" />
						<label for="element_2_1">MM</label>
					</span>
					<span>
						<input id="element_2_2" name="element_2_2" class="element text" size="2" maxlength="2" value="" type="text" />
						<label for="element_2_2">DD</label>
					</span>
					<span>
						<input id="element_2_3" name="element_2_3" class="element text" size="4" maxlength="4" value="" type="text" />
						<label for="element_2_3">YYYY</label>
					</span>
				
					<span id="calendar_2">
						<img id="cal_img_2" class="datepicker" src="calendar.gif" alt="Pick a date." />	
					</span>
					<script type="text/javascript">
						Calendar.setup({
						inputField	 : "element_2_3",
						baseField    : "element_2",
						displayArea  : "calendar_2",
						button		 : "cal_img_2",
						ifFormat	 : "%B %e, %Y",
						onSelect	 : selectDate
						});
					</script>
				</li>		
				<li id="li_3" >
		<label class="description" for="element_3">Meeting Time </label>
		<span>
			<label>HH</label>		
			<select id="element_3_1" name="element_3_1" class="element text" >
				<?php
					for($i=0; $i<12 ; ++$i)
					{
						if($i<9)
						{
							$j = $i + 1;
							echo "<option value=\\"0" . $j . "\\">0" .$j . "</option>";
						}
						else
						{
							$j = $i + 1;
							echo "<option value=\\"" . $j . "\\">" .$j . "</option>";
						}
					}
				?>
				
			</select>
		</span>
		<span>
			<label>MM</label>		
			<select id="element_3_2" name="element_3_2" class="element text" >
			<option value = "00" selected = "selected">00</option>
			<option value = "15">15</option>
			<option value = "30">30</option>
			<option value = "45">45</option>
			</select>
		</span>
		<span>
			<label>AM/PM</label>		
			<select class="element select" id="element_3_4" name="element_3_4">
				<option value="AM" >AM</option>
				<option value="PM" >PM</option>
			</select>

		</span>
		</li>		<li id="li_4" >
		<label class="description" >Meeting Place </label>
		<div>
		in meeting place div
			<select>
			<option value="0">Please Select</option>
<?php
 		  $myQuery = "SELECT * FROM meeting_places ";
	
		$result = mysql_query($myQuery);

		while ($info = mysql_fetch_assoc($result))
		{
			echo '<option value="' . $info['id'] . '">' . $info['MeetingPlace'] . '</option>';
		}

?>			
			</select>
		</div>
		</li>		
		<li id="li_5" >
		<label class="description" >Meeting Address </label>
		
		<div>
			<input id="element_5_1" name="element_5_1" class="element text large" type="text" />
			<label for="element_5_1">Street Address</label>
		</div>
	
		<div>
		<span style="float:left">
			<input id="element_5_3" name="element_5_3" class="element text medium" style="width:14em" type="text" />
			<label for="element_5_3">City</label>
		</span>
		<span style="float:left">

			<select id="element_5_4" name="element_5_4" class="element text medium" type="text" >

				<option value="ID">ID</option>
				<option value="WA">WA</option>
			</select>
			<label for="element_5_4">State</label>			
		</span>
		<span style="float:left">
			<input id="element_5_5" name="element_5_5" class="element text medium" type="text" />
			<label for="element_5_5">Zip Code</label>
			</span>
		</div>
	
		</li>		
		
		<li id="li_6" >
		<label class="description" for="element_6">Comments </label>
		<div>
			<textarea id="element_6" name="element_6" class="element textarea medium" rows="4" cols="50"></textarea>
		</div>
		</li>
			
					<li class="buttons">
			    <input type="hidden" name="form_id" value="754506" />
			    <input id="newForm" method="post" class="button_text" type="submit" name="addNew" value="Add New" action="<?= $_SERVER['PHP_SELF'] ?>"/>
				<input id="saveForm" method="post" class="button_text" type="submit" name="submit" value="Submit" action="<?= $_SERVER['PHP_SELF'] ?>"/>
		</li>
			</ul>
		</form>	
		<?php }
		?>		
	</div>
	
    <script type="text/javascript" src="http://ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
    <script type="text/javascript">
      function insertResults(json){
        $("#element_5_1").val(json["StreetAddress"]);
        $("#element_5_3").val(json["City"]);
        $("#element_5_4").val(json["State"]);
        $("#element_5_5").val(json["ZipCode"]);
		$("#element_6").val(json["Comments"]);
      }

      function clearForm(){
        $("#element_5_1, #element_5_3, #element_5_4, #element_5_5, #element_6").val("");
      }

      function makeAjaxRequest(placeId){
        $.ajax({
          type: "POST",
          data: { placeId: placeId },
          dataType: "json",
          url: "process_ajax999.php",
          success: function(json) {
            insertResults(json);
          }
        });
      }

      $("#meetingPlace").on("change", function(){
        var id = $(this).val();
        if (id === "0"){
		alert('id is zero');
          clearForm();
        } else {
          makeAjaxRequest(id);
        }
      });
    </script>
	</body>
</html>

**Comment by NoliJay** [comment]
<?php
	// Connection Variables
	$hostname="localhost";
	$mysql_login="user";
	$mysql_password="pw";
	$database='sitepoint';
	

	// Connect to the database server
	$link = mysql_connect($hostname, $mysql_login, $mysql_password);
	if (!$link)
	{
		die('Could not connect: ' . mysql_error());
	}
   $db_selected = mysql_select_db($database, $link);
  if (!$db_selected) {
      die ('Can\\'t use ' . $database . ' : ' . mysql_error());
  }
  $placeId = $_POST['placeId'];

  $query = "SELECT * from meeting_places";
  $result = mysql_query($query) or die('Query failed: ' . mysql_error());
 echo 'in process_ajax99.php<br>';
	while ($row = mysql_fetch_assoc($result)) {
    if ($placeId == $row['id']){
      echo json_encode($row);
    }
  }
?>

**Comment by pash0107** [comment]
function get_print_subjects($id){
	$totalunits = 0;
	$sel = mysql_query("SELECT * FROM year_level_subject as yls join subjects as s on yls.subj_id=s.subj_id  where yls.year_id='$id' ");
	while($row=mysql_fetch_assoc($sel)){
		extract($row);
		echo '<tr>';
		echo '<td>'.$subj_code.'</td>';
		echo '<td>'.$subj_desc.'</td>';
		echo '<td>'.$units.'</td>';
		echo '</tr>';
		$totalunits += $units;
	}
	echo '<tr>';
	echo '<td></td>';
	echo '<td></td>';
	echo '<td></td>';
	echo '</tr>';

	echo '<tr>';
	echo '<th>Total Units</th>';
    echo '<td></td>';
	echo '<th>'.$totalunits.' units'.'</th>';
	echo '</tr>';
}
